packages/mask/src/components/shared/SelectRecipients/useContacts.ts

```diff
@@ -0,0 +1,28 @@
+import { useAsyncRetry } from 'react-use'
+import type { ProfileInformation } from '@masknet/shared-base'
+import type { AsyncStateRetry } from 'react-use/lib/useAsyncRetry.js'
+import { useCurrentPersona } from '../../DataSource/usePersonaConnectStatus.js'
+import Services from '../../../extension/service.js'
+
+export function useContacts(network: string): AsyncStateRetry<ProfileInformation[]> {
+    const currentPersona = useCurrentPersona()
+
+    return useAsyncRetry(async () => {
+        const values = await Services.Identity.queryRelationPaged(
+            currentPersona?.identifier,
+            {
+                network,
+                pageOffset: 0,
+            },
+            1000,
+        )
+        if (values.length === 0) return []
+
+        const identifiers = values.map((x) => x.profile)
+        const [, profiles] = await Promise.all([
+            Services.Identity.queryAvatarsDataURL(identifiers),
```

Comment:
The result of `queryAvatarsDataURL` didn't use. But it seems not a side-effect method according to its name.